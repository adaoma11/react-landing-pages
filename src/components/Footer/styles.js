import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
