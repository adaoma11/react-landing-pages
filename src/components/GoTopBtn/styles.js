import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
