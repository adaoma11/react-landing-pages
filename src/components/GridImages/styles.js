import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
