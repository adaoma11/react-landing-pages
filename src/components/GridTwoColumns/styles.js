import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
