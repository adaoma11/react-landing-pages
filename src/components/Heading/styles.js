import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, $dark, size, $upperCase }) => css`
    color: ${$dark ? theme.colors.primary : theme.colors.white};

    font-size: ${theme.fonts.sizes[size]};

    ${size == 'xlarge' &&
    css`
      @media ${theme.medias.small} {
        font-size: ${theme.fonts.sizes.large};
      }
    `}

    text-transform: ${$upperCase ? 'uppercase' : 'none'}
  `}
`;
