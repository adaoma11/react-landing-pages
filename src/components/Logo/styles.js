import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.heading};
    font-size: ${theme.fonts.sizes.large};
    font-weight: bold;
    color: ${theme.colors.primary};
    text-decoration: none;

    > img {
      min-height: 40px;
      max-height: 50px;
      max-width: 150px;
      min-width: 40px;
    }
  `}
`;
