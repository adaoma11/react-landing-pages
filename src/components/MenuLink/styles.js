import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    padding: 15px;
    font-family: ${theme.fonts.family.default};
    font-size: ${theme.fonts.sizes.small};
    color: ${theme.colors.primary};
    text-decoration: none;
    position: relative;

    &::after {
      content: '';
      position: absolute;
      background: ${theme.colors.secondary};
      left: 25%;
      bottom: 0;
      width: 50%;
      height: 3px;
      transform: scale(0);
      transition: all 0.4s ease-in-out;
    }

    &:hover::after {
      transform: scale(1);
    }
  `}
`;
