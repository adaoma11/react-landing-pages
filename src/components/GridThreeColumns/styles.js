import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
