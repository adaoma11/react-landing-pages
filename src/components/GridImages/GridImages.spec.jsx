import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { GridImages } from '.';

describe('<GridImages />', () => {
  const mockData = {
    title: 'Grid Images Example',
    description: 'Uma breve descrição sobre este album',
    gridItems: [
      {
        id: 1,
        altText: 'Imagem 1',
        imgSrc: 'https://picsum.photos/300?random=1',
      },
      {
        id: 2,
        altText: 'Imagem 2',
        imgSrc: 'https://picsum.photos/300?random=2',
      },
      {
        id: 3,
        altText: 'Imagem 3',
        imgSrc: 'https://picsum.photos/300?random=3',
      },
      {
        id: 4,
        altText: 'Imagem 4',
        imgSrc: 'https://picsum.photos/300?random=4',
      },
      {
        id: 5,
        altText: 'Imagem 5',
        imgSrc: 'https://picsum.photos/300?random=5',
      },
      {
        id: 6,
        altText: 'Imagem 6',
        imgSrc: 'https://picsum.photos/300?random=6',
      },
    ],
    hasBg: false,
    titleUpperCase: false,
  };

  it('should render component properly', () => {
    renderTheme(<GridImages {...mockData} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('paragraph')).toBeInTheDocument();
    expect(screen.getAllByRole('img')).toHaveLength(mockData.gridItems.length);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<GridImages {...mockData} />);
    expect(container).toMatchSnapshot();
  });
});
