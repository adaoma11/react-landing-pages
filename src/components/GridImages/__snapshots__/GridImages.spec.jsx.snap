// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImages /> should match snapshot 1`] = `
.c5 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
}

.c6 {
  overflow: hidden;
}

.c7 {
  width: 100%;
  transition: all 0.5s ease-in-out;
}

.c7:hover {
  transform: scale(1.2) rotate(5deg);
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: 100vh;
  padding: 50px 0;
  background-color: #FFFFFF;
}

.c1 {
  max-width: 1200px;
  width: 100%;
  padding: 30px;
}

.c2 {
  margin-bottom: 30px;
}

.c3 {
  color: #0A1128;
  font-size: 3rem;
  margin-bottom: 1rem;
  text-transform: none;
  text-align: center;
}

.c4 {
  font-family: 'Poppins',sans-serif;
  font-size: 1.5rem;
  color: #0A1128;
  text-align: center;
  font-weight: normal;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 2rem;
  }
}

<div>
  <section
    class="c0"
  >
    <article
      class="c1"
    >
      <header
        class="c2"
      >
        <h2
          class="c3"
        >
          Grid Images Example
        </h2>
        <p
          class="c4"
        >
          Uma breve descrição sobre este album
        </p>
      </header>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <img
            alt="Imagem 1"
            class="c7"
            src="https://picsum.photos/300?random=1"
          />
        </div>
        <div
          class="c6"
        >
          <img
            alt="Imagem 2"
            class="c7"
            src="https://picsum.photos/300?random=2"
          />
        </div>
        <div
          class="c6"
        >
          <img
            alt="Imagem 3"
            class="c7"
            src="https://picsum.photos/300?random=3"
          />
        </div>
        <div
          class="c6"
        >
          <img
            alt="Imagem 4"
            class="c7"
            src="https://picsum.photos/300?random=4"
          />
        </div>
        <div
          class="c6"
        >
          <img
            alt="Imagem 5"
            class="c7"
            src="https://picsum.photos/300?random=5"
          />
        </div>
        <div
          class="c6"
        >
          <img
            alt="Imagem 6"
            class="c7"
            src="https://picsum.photos/300?random=6"
          />
        </div>
      </div>
    </article>
  </section>
</div>
`;
